pacu/modules/eks__enum/main.py

```diff
@@ -0,0 +1,85 @@
+parser.add_argument('--all', required=False, action='store_true', default=False, help='Enumerate all EKS resources.')
+
+def main(args, pacu_main):
+    args = parser.parse_args(args)
+    print = pacu_main.print
+    session = pacu_main.get_active_session()
+    get_regions = pacu_main.get_regions
+    data = {}
+    cluster_count = 0
+
+    if args.regions is None:
+        regions = get_regions('eks')
+        if regions is None or regions == [] or regions == '' or regions == {}:
+            print('This module is not supported in any regions specified in the current sessions region set. Exiting...')
+            return
+    else:
+        regions = args.regions.split(',')
+    for region in regions:
+        eks_client = pacu_main.get_boto3_client('eks', region)
+        clusters = eks_client.list_clusters()["clusters"]
+        print(f"clusters in {region}: {clusters}")
+        if len(clusters) > 0:
+            cluster_count += len(clusters)
+            data[region] = {
+            "clusters": {}
+            }
+            for cluster in clusters:
+                if args.verbose:
+                    data[region]['clusters'][cluster] = {
+                        "cluster_description": eks_client.describe_cluster(name=cluster)["cluster"],
+                        "nodegroups": eks_client.list_nodegroups(clusterName=cluster)["nodegroups"]
+                    }
+                    if args.addons or args.all:
+                        data[region]['clusters'][cluster]["addons"] = eks_client.list_addons(clusterName=cluster)["addons"]
+                    if args.fargate_profiles or args.all:
+                        data[region]['clusters'][cluster]["fargate_profiles"] = eks_client.list_fargate_profiles(clusterName=cluster)["fargateProfileNames"]
+                    if args.identity_provider_configs or args.all:
+                        data[region]['clusters'][cluster]["identity_provider_configs"] = eks_client.list_identity_provider_configs(clusterName=cluster)["identityProviderConfigs"]
+                else:
+                    data[region]['clusters'][cluster] = {
+                        "nodegroups": eks_client.list_nodegroups(clusterName=cluster)["nodegroups"]
+                    }
+                    if args.addons or args.all:
+                        data[region]['clusters'][cluster]["addons"] = eks_client.list_addons(clusterName=cluster)["addons"]
+                    if args.fargate_profiles or args.all:
+                        data[region]['clusters'][cluster]["fargate_profiles"] = eks_client.list_fargate_profiles(clusterName=cluster)["fargateProfileNames"]
+                    if args.identity_provider_configs or args.all:
+                        data[region]['clusters'][cluster]["identity_provider_configs"] = eks_client.list_identity_provider_configs(clusterName=cluster)["identityProviderConfigs"]
```

Comment:
These all need pagination, otherwise we'll only see partial results in some cases. Looks like the ECS resource has paginators for this, so that may help here (https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/eks.html#paginators).

Reply:
Sounds good. Working on pagination now.

Reply:
Pagination should be implemented with this last commit. Running some rudimentary testing now.